src/main/java/com/myhome/controllers/HouseController.java

```diff
@@ -156,4 +158,36 @@ public HouseController(CommunityHouseRepository communityHouseRepository,
       return ResponseEntity.status(HttpStatus.NOT_FOUND).build();
     }
   }
+  @Operation(description = "Post Date Interval for houseId and Member Id")
+  @PostMapping(
+          path = "/house/member/interval",
+          produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE},
+          consumes = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<HouseHistory> postInterval(@Valid @RequestBody HouseHistoryDto houseHistoryDto) {
+    log.trace("Received request to postinterval reuqest "+ CommonHelper.getStringFromObject(houseHistoryDto));
+
+    HouseHistory houseHistory = houseService.addInterval(houseHistoryDto);
+    if(!CommonHelper.empty(houseHistory) ){
+      return ResponseEntity.status(HttpStatus.CREATED).body(houseHistory);
+    }
+
+    return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(houseHistory);
+  }
+
+  @Operation(description = "List History of House/Member")
+  @GetMapping(
+          path = "/house/history/{houseId}",
+          produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_XML_VALUE}
+  )
+  public ResponseEntity<List<HouseHistoryDto>> getHouseHistory(@PathVariable(required = true) String houseId,
+                                                               @RequestParam(required = false) String memberId) {
+    log.trace("Received request to get history of a house with id[{}] and member Id with id[{}]", houseId,memberId);
+    if (!houseService.getHouseDetailsById(houseId).isPresent()) {
+      return ResponseEntity.status(HttpStatus.NOT_FOUND).body(new ArrayList<HouseHistoryDto>());
+    }
+    List<HouseHistory> houseHistoryPOList = houseService.getHouseHistory(houseId,memberId);
```

Comment:
What is `PO` ?

Reply:
Persistent Object..

Reply:
Is it needed to be included here? If so why the name is not `houseHistoryPersistentObjectList`?
I would leave it as `houseHistory`.